core/src/main/java/com/netflix/iceberg/util/BinPacking.java

```diff
@@ -107,7 +115,7 @@ public boolean hasNext() {
           bins.addLast(bin);
 
           if (bins.size() > lookback) {
-            return ImmutableList.copyOf(bins.removeFirst().items());
+            return ImmutableList.copyOf((largestBinFirst ? getLargestBin() : bins.removeFirst()).items());
```

Comment:
It would be more clear to separate this into an `if`/`else` instead of using a ternary operator:

```java
Bin bin;
if (largestBinFirst) {
  bin = removeLargestBin(bins);
} else {
  bin = bins.removeFirst();
}
return ImmutableList.copyOf(bin.items());
```